hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/util/TestHttpExceptionUtils.java

```diff
@@ -164,4 +164,30 @@ public void testValidateResponseJsonErrorUnknownException()
     }
   }
 
+  @Test
+  public void testValidateResponseJsonErrorNonException() throws IOException {
+    Map<String, Object> json = new HashMap<String, Object>();
+    json.put(HttpExceptionUtils.ERROR_EXCEPTION_JSON, "invalid");
+    // test case where the exception classname is not a valid exception class
+    json.put(HttpExceptionUtils.ERROR_CLASSNAME_JSON, String.class.getName());
+    json.put(HttpExceptionUtils.ERROR_MESSAGE_JSON, "EX");
+    Map<String, Object> response = new HashMap<String, Object>();
+    response.put(HttpExceptionUtils.ERROR_JSON, json);
+    ObjectMapper jsonMapper = new ObjectMapper();
+    String msg = jsonMapper.writeValueAsString(response);
+    InputStream is = new ByteArrayInputStream(msg.getBytes());
+    HttpURLConnection conn = Mockito.mock(HttpURLConnection.class);
+    Mockito.when(conn.getErrorStream()).thenReturn(is);
+    Mockito.when(conn.getResponseMessage()).thenReturn("msg");
+    Mockito.when(conn.getResponseCode()).thenReturn(
+            HttpURLConnection.HTTP_BAD_REQUEST);
+    try {
+      HttpExceptionUtils.validateResponse(conn, HttpURLConnection.HTTP_CREATED);
```

Comment:
use `LambdaTestUtils.intercept()`, which is based off ScalaTest's version, includes test for error message text.

```java
intercept(IOException.class,
  "HTTP status [400], exception [java.lang.String], message [EX], URL [null]",
  () -> HttpExceptionUtils.validateResponse(conn, HttpURLConnection.HTTP_CREATED);
```

  
```

Reply:
I can change this but I just want to highlight that this is a modified copy of existing tests in this test class and this is the code style in the existing tests.

Reply:
old code, going near the test is the time to clean it up.